src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -270,23 +293,75 @@ def _get_partitioned_plan(self, root: TreeNode, child: TreeNode) -> SnowflakePla
 
         return temp_table_plan
 
-    def _is_node_valid_to_breakdown(self, node: LogicalPlan) -> Tuple[bool, int]:
+    def _is_node_valid_to_breakdown(
+        self, node: TreeNode, root: TreeNode
+    ) -> Tuple[Optional[CompilationStageTelemetryField], int]:
         """Method to check if a node is valid to breakdown based on complexity score and node type.
 
         Returns:
-            A tuple of boolean indicating if the node is valid for partitioning and the complexity score.
+            A tuple of =>
+                CompilationStageTelemetryField: indicating the primary reason
+                    for invalidity if the node is not valid else None.
+                int: the complexity score of the node.
         """
         score = get_complexity_score(node)
-        valid_node = (
+        is_valid = True
+        reason_if_invalid = None
+        if not (
             self.complexity_score_lower_bound
             < score
             < self.complexity_score_upper_bound
-        ) and self._is_node_pipeline_breaker(node)
-        if valid_node:
+        ):
+            is_valid = False
+            reason_if_invalid = CompilationStageTelemetryField.INVALID_DUE_TO_SCORE
```

Comment:
let's get enum class about all reasons, and we can we can have more complete reason, for this one, we can even separate them to SCORE_LOW,  and SCORE_HIGH